File: public/codesamples/posts/graphes_connections/Connection.cs
namespace Posts.GrahpesConnections
{
  public class Connection
  {
    private readonly Node _from;
    public Node From {
      get { return _from; }
    }
    private readonly Node _to;
    public Node to
    {
      get { return to; }
    }
    private float _weight;
    public float Weight
    {
      get { return _weight; }
      set { _weight = value; }
    }
    public Connection(Node from, Node to, float weight)
    {
      this._from = from;
      this._to = to;
      this._weight = weight;
    }
  }
}

File: public/codesamples/posts/graphes_connections/Connection.java
package posts.graphes_connections;

public class Connection {
  private Node from;
  private Node to;
  private float weight;

  public Connection(Node from, Node to, float weight) {
    this.from = from;
    this.to = to;
    this.weight = weight;
  }

  public Node getFrom() {
    return from;
  }

  public Node getTo() {
    return to;
  }

  public float getWeight() {
    return weight;
  }

  public void setWeight(float weight) {
    this.weight = weight;
  }
}
